docs/development_workflow.rst

```diff
@@ -96,6 +96,25 @@ First, all automated checks need to pass before merging. Then...
 * Stale reviews should be dismissed by the PR submitter when the feedback has been addressed
 
 
+Copyright and licensing
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The project as a whole is released under the MIT license, and copyright on its code is held by multiple parties including Learning Equality.
+
+Individual files, such as code copied in from other projects may be under a different license, if that license is compatible.
+
+Similarly, files from Kolibri may end up being copied into other projects.
+
+For these reasons, it’s encouraged to list copyright and license data explicitly at the top of each file. For example:
```

Comment:
> For these reasons, it’s encouraged to list copyright and license data explicitly at the top of each file.

We don't feel that we'll actually encourage this and retain this as a convention. We would like to support this situation, but without additional hurdles and conventions. So I would like to request some rephrasing here.

There seem to be 2 primary situations:
- vendored code snippets from another already licensed project
- new code where the creators desire this description

For vendored code snippets, it's absolutely appropriate to include copyright and licensing declarations within the file or alongside the code that was copied. We could, and probably should, mention that here.

For new code, and perhaps even whole complete vendored files, I suggest that we take a centralized metadata approach that the REUSE specification mentions, by having a file `.reuse/dep5` which itemizes each of these declarations.
```
Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
Upstream-Name: kolibri
Upstream-Contact: Learning Equality <info@learningequality.org>
Source: https://github.com/learningequality/kolibri

Files: path/to/file
Copyright: 2023 Ann Contributor
License: MIT
```

Reply:
Thanks for taking the time to give this some thought. I had meant these changes as a fairly low-key way of documenting how to put license/copyright data in a file header if people wanted to, rather than a below-the-radar attempt to significantly change the norms/practices for Kolibri. The scope I had in mind for this was only for future contributions to Kolibri, and not retrospectively adding copyright/license headers to existing files.

I think it would be great if someone _did_ retrospectively add headers to existing files, but it’s a *lot* of work and imo only worthwhile when there’s a strong use case for sorting it out — perhaps if a prospective big government user of Kolibri had software licensing reporting requirements to meet before they could use it. That kind of use case is one of the motivations for REUSE existing. The work comes from doing the git archaeology to work out the copyright holders of each file. In another project I work on [we’ve been slowly chipping away at this](https://gitlab.gnome.org/GNOME/glib/-/issues/1415) over time, with CI in place to never allow the REUSE compliance level (as reported by `reuse lint`) to decrease. I am not proposing that here now though.

REUSE allows for using `.reuse/dep5`, but it’s mildly [discouraged](https://reuse.software/faq/#bulk-license) compared to putting the information in the file headers where possible. My take on it is that the use of a `dep5` file keeps the license/copyright separate from the code, so it doesn’t solve the problem of that metadata going missing if someone copies code out of Kolibri into another project. It also makes it less likely that people will add themselves to the copyright for a file when they make a copyrightable contribution to it, because that metadata will be far away from what they’re editing. That reduces the chance of the copyright listings staying up to date.

Using a `dep5` file is an improvement on freeform documentation of the license in `LICENSE`, though, as it is machine readable. It can also be iterated on in future to refine the copyright listings.

I can completely appreciate you don’t want to change existing working practices, or have another thing to remember to remind people about when reviewing PRs. How about changing the wording in this PR from “it’s encouraged to list copyright and license data explicitly” to “it’s supported to list copyright and license data explicitly”? For the reasons above, I think steering people towards `dep5` is a red herring.

Reply:
All that said, if you think this discussion is not worth spending time on (which I can appreciate), then please say, and I’ll drop the `development_workflow.rst` changes from the PR entirely and we can just go with the first commit about “and other contributors” for now. :)

Reply:
I read some background on copyrights, and have come to a conclusion which I think is inline with most of your changes, but also supports the approach we're preferring. 

First, with respect to the Kolibri project as a whole, it is true that any 'material contribution' to the project should be individually copyrightable by the author making the contribution. If it isn't copyrightable by them, then that falls into the category of code that should clearly declare the copyright holder and the license that allows it as a contribution. So I think the meaningful aspect here is that distinction. 

Additionally, the project itself should be viewed as Joint Authorship, in that all contributors gain equal copyright on the resulting work, because the intent of most contributions is to create the single application, Kolibri. Therefore, tracking the individual copyright holder per file becomes unnecessary, because any originally authored contribution with that intent makes the contributor a joint author of the entire work. An originally authored contribution is and should be copyrightable, but as a contribution to a larger work, any presumed rights of that copyright aren't meaningful because joint authors of Kolibri also have equal rights. Anyone reusing bits of code from Kolibri should specify all joint authors because of this, by referencing the entire project's license and authorship information.

Furthermore, as an open source project, and a learning platform, we're definitely willing to recommend and advocate for best practices regarding code that is not originally authored by the contributor-- it should appropriately attribute its source and license. Framing it that way in our docs makes a lot of sense, since our vision is that 'all learners [including the many junior, external contributors we involve] develop their own agency, create positive transformation, and flourish', which ideally results in positive change in open-source software development in regards to these challenges.

Lastly, if a contribution is somewhat standalone, in that the intent doesn't directly develop the single work itself, then it is appropriate to document copyrights and licensing explicitly. This can be the case for many auxiliary Kolibri plugins that do not serve to complete it as a learning platform application, but only add-on to its functionality at run time. Kolibri plugins can be distributed in separate repositories, and under separate licensing, which should likely be the case for those introducing auxiliary functionality.

Reply:
It looks like you’ve put quite some time into looking into this! Sorry for my delayed reply. I am not a lawyer, so I’m going on the basis of [this page](https://opensource.stackexchange.com/questions/12439/does-contributing-to-software-project-make-it-joint-authorship-or-a-collective-w) and its explanations of joint authorship.

Joint authorship is, in my experience, not the normal way of approaching free software copyright. From what I’ve read, it seems to be considered a valid way of managing the copyrights, but certainly not the standard way.

Given that, and the fact that I’m not a lawyer and not familiar with the full implications of joint authorship, I’m going to trim down my proposed changes in this PR so that it can get unstuck.

Given that you’ve done the research on copyright and seem to have come to a position you’re happy with, it seems like an easy next step would be for you to modify the documentation a bit further to encode that position, if you want to make it official. Personally I’d be cautious about taking a non-traditional (for FOSS) position on copyright ownership, but it’s not my project and I may well be being over-cautious because I’m not a lawyer. :)

Reply:
I'm not a lawyer either, but as the link you shared points out in regards to categorizing multiple contributors and their copyrights, it's mostly 'Joint authorship' vs 'Collective work', and 'Collective work' simply doesn't apply since contributions are generally not standalone. With the perspective of the entire project in mind, not just individual contributors, it makes sense this way. While a contributor holds copyrights to their contributions, it isn't really materially meaningful because again, the contributions are likely not standalone and were developed with the intent to work in conjunction with the project itself, inevitably requiring that the contributor understands the limitations of those rights in regards to the project's licensing.

Overall, I feel we strive to prioritize the 'open' aspect of our projects and our product ecosystem, and avoid any attempts to elicit control through legal avenues. Your improvements to making the value of contributions clear is definitely worthwhile though.

Thanks for making the updates. I will give another review.